packages/eslint-plugin-svelte/src/shared/svelte-compile-warns/index.ts

```diff
@@ -407,6 +407,7 @@ function getWarningsFromCode(
 } {
 	try {
 		const result = compiler.compile(code, {
+			...context.sourceCode.parserServices.svelteParseContext?.svelteConfig?.compilerOptions,
```

Comment:
To avoid unintended behavior, can you please pass only the necessary options?

Since `generate: false` is different from the compiler used by the user, I think it is highly likely that the user's defined `compilerOptions` cannot be used as is.

Reply:
Compiler options have many properties, and I don't think we understand how they affect linting. Also, I don't think users set compiler options with linting in mind.
Therefore, I think it's better to avoid passing them directly.
I think it's better to pass only the necessary properties that we understand.
https://svelte.dev/docs/svelte/svelte-compiler#CompileOptions
https://svelte.dev/docs/svelte/svelte-compiler#ModuleCompileOptions

Reply:
> I don't think we understand how they affect linting

I believe I understand your concern. I have no objection to making the fix.
However, in this particular case, do we really need to worry about the affect? Wouldn’t it be fine to simply compile exactly as the user specified and then display whatever warnings are generated to the user? If something strange happens, wouldn’t that just mean the user’s configuration was wrong?

Reply:
The reason I'm concerned may be because I don't know the internals of the compiler, so I don't have the information to not worry 😓